### PR TITLE
Fix sign duping

### LINKED ISSUE #8901: You can duplicate SECURE AREA signs
(WEB REPORT BY: zealotron REMOTE: 158.69.55.2:7777)

> Revision (Should be above if you're viewing this from ingame!)
> 62a642dc363f8826adee1e97ef0e13713d3d71b5 
> 
> General description of the issue
> You can duplicate the SECURE AREA signs
> 
> What you expected to happen
> 
> What actually happened
> 
> Steps to reproduce if possible
> 1. Find a sign
> 2. REmove it with a screwdriver
> 3. Spam click it with a screwdriver to re apply, go through all the prompts.
> 4. You get a new sign for every prompt you get

### PR DESCRIPTION
**Pull request by gbasood**
Fixes #8901

### CODE REVIEW

File: code/game/objects/structures/signs.dm [hunk 1]
```diff
@@ -59,7 +59,7 @@
 /obj/item/sign/attackby(obj/item/tool as obj, mob/user as mob)	//construction
 	if(isscrewdriver(tool) && isturf(user.loc))
 		var/direction = input("In which direction?", "Select direction.") in list("North", "East", "South", "West", "Cancel")
-		if(direction == "Cancel") return
+		if(direction == "Cancel" || src.loc == null) return // We can get qdel'd if someone spams screwdrivers on signs before responding to the prompt.
```
Comment (clusterfack) [inline code/game/objects/structures/signs.dm]: There's no do_after here, how could they spam it?

Comment (gbasood) [inline code/game/objects/structures/signs.dm:62]: If they spam a screwdriver before responding to prompts they can duplicate the prompt and make multiple new signs.

(READ THE ISSUE GEEZ)

Comment (clusterfack) [inline code/game/objects/structures/signs.dm]: never!

**Comment by gbasood** [comment]
~~Actually in theory you shouldnt be able to get more than one sign because the ones after the first should be nullspaced
I'll look at this later~~

``` DM
var/obj/structure/sign/S = new(user.loc)
```

Turns out I was just tired.

**Pull request merged**